decompiler/IR2/OpenGoalMapping.cpp

```diff
@@ -11,13 +11,12 @@ typedef OpenGOALAsm::InstructionModifiers MOD;
 const std::map<InstructionKind, OpenGOALAsm::Function> MIPS_ASM_TO_OPEN_GOAL_FUNCS = {
     // ----- EE -------
     // Instructions that are nopped
-    //    {InstructionKind::MTC0,
-    //     {".nop", {MOD::SKIP_IT}}},  // they only use this for performance counters / Count /
-    //     Debug
-    //    {InstructionKind::MTPC, {".nop", {MOD::SKIP_IT}}},
-    //    {InstructionKind::MFPC, {".nop", {MOD::SKIP_IT}}},
-    //    {InstructionKind::SYNCP, {".nop", {MOD::SKIP_IT}}},
-    //    {InstructionKind::SYNCL, {".nop", {MOD::SKIP_IT}}},
+    // they only use this for performance counters / Count / Debug
+    {InstructionKind::MTC0, {".nop", {MOD::SKIP_IT}}},
+    {InstructionKind::MTPC, {".nop", {MOD::SKIP_IT}}},
+    {InstructionKind::MFPC, {".nop", {MOD::SKIP_IT}}},
+    {InstructionKind::SYNCP, {".nop", {MOD::SKIP_IT}}},
```

Comment:
I'd rather keep these in if possible `.sync.l`/`.sync.p` is not just for performance counters, `mtc0` is for all different things.

In `gcommon.gc` there is a "fake assembly" thing that you can use to make the compiler ignore the assembly, which I'd rather use:
```
;;;;;;;;;;;;;;;;;;;;;;;
;; PC Port asm macros
;;;;;;;;;;;;;;;;;;;;;;;
(#when PC_PORT
  ;; SYNC is an EE instruction that waits for various memory access and DMA to be completed
  ;; DMA will be instant in the PC port, so these are no longer necessary
  (fake-asm .sync.l)
  (fake-asm .sync.p)
  ;; Copies the contents of a cop0 (system control) register to a gpr
  (fake-asm .mfc0 dest src)
  ;; Copies the contents of a gpr to a cop0 (system control) register
  (fake-asm .mtc0 dest src)
  )
```
Maybe we can add these here and also to `decompiler-macros.gc` (I think I forgot this)